cmd/retry.go

```diff
@@ -91,16 +91,26 @@ func newRetryTimer(unit time.Duration, cap time.Duration, jitter float64, doneCh
 	go func() {
 		defer close(attemptCh)
 		nextBackoff := 0
+		// Channel used to signal after the expiry of backoff wait seconds.
+		var timer *time.Timer
 		for {
-			select {
-			// Attempts starts.
+			select { // Attempts starts.
 			case attemptCh <- nextBackoff:
 				nextBackoff++
+				timer = time.NewTimer(exponentialBackoffWait(nextBackoff))
 			case <-doneCh:
 				// Stop the routine.
 				return
 			}
-			time.Sleep(exponentialBackoffWait(nextBackoff))
+			// wait till next backoff time or till doneCh gets a message.
+			select {
+			case <-timer.C:
+			case <-doneCh:
+				timer.Stop()
+				// stop the go routine and return.
+				return
```

Comment:
nit: stop the timer and return.
... and the comment should appear before the code.